homeassistant/components/bluetooth/__init__.py

```diff
@@ -227,14 +254,21 @@ def _device_detected(
     ) -> None:
         """Handle a detected device."""
         matched_domains: set[str] | None = None
-        if device.address not in self._matched:
+        manufacturer_ids = list(advertisement_data.manufacturer_data).sort()
+        device_matcher = (device.address, manufacturer_ids)
```

Comment:
In fjäråskupan, the manufacture data changes all the time. Its the current state of the device. So its not an ID

Reply:
I'll probably end up reverting this as I think it was actually just a bug in the govee parser

Reply:
Reverted in 6b640757f7cb4efe11164bc23b45f36c0ac570d4

Reply:
Sometimes `manufacturer_data` isn't populated right away. So we still need to dispatch discovery flows again if we see the device a another time and it has `manufacturer_data` but never did before

`2022-07-09 21:14:17.477 DEBUG (MainThread) [homeassistant.components.inkbird.data] Parsing inkbird BLE advertisement data: BluetoothServiceInfo(name='sps', address='5C87D953-8E5A-FE7B-F04B-559AADE4E365', rssi=-71, manufacturer_data={}, service_data={}, service_uuids=['0000fff0-0000-1000-8000-00805f9b34fb'])`